### PR TITLE
feat(condo): DOMA-5750 search billing receipts without consumer service

### PR DESCRIPTION
**Pull request by ekabardinsky**

### CODE REVIEW

File: apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Generated by `createservice billing.SearchBillingReceiptsWithoutConsumerService --type queries`
+ */
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+
+async function canSearchBillingReceiptsWithoutConsumer ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    return true
```
Comment (pahaz) [inline apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js:10]: O-o!

**Review by pahaz — CHANGES_REQUESTED** [review]
It's strange to give access to anyone to all Receipts

File: apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js (hunk 1 above)
Comment (akarjakin) [reply apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js:10]: any non-deleted user will be able to read these... are we sure about that ?

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                    } = residentsWithoutConsumer[i]
+
+                    const property = await Property.getOne(context, { id: propertyId, deletedAt: null }, {
+                        doesNotExistError: errors.MISSING_RESIDENT_PROPERTY,
+                    })
+
+                    // assemble conditions
+                    const addressCondition = {
+                        OR: [
+                            { property: { address: property.address } },
```
Comment (akarjakin) [inline apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js:117]: address_i:

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                            { property: { addressKey: property.addressKey } },
+                            { property: { normalizedAddress: property.address } },
```
Comment (akarjakin) [inline apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js:119]: normalizedAddress_i:

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                            unitType,
+                        },
+                    }
+                    const integrationCondition = {
+                        context: {
+                            status: CONTEXT_FINISHED_STATUS,
+                        },
+                    }
+
+                    const receipts = await BillingReceipt.getAll(context, {
```
Comment (akarjakin) [inline apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js:134]: if there are lots of receipts, better to use something like loadListByChunks or add pagination

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                    user: { id: context.authedItem.id },
+                    deletedAt: null,
+                })
+
+                if (residents.length !== residentIds.length) {
+                    throw new GQLError(errors.MISSING_RESIDENTS, context)
+                }
+
+                // retrieve related ServiceConsumers
+                const consumers = await ServiceConsumer.getAll(context, {
```
Comment (akarjakin) [inline apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js:88]: loadListByChunks

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js (hunk 2 above)
Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: 10x

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js (hunk 3 above)
Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: 10x

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js (hunk 4 above)
Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: Now it becomes BillingAccount entity searching only for newly registered of condo platform. It is not a big result we are expecting as a result.

Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: But probably you are right, better to support all cases

Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: fixed

File: apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js (hunk 5 above)
Comment (ekabardinsky) [reply apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js]: fixed

File: apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js (hunk 1 above)
Comment (ekabardinsky) [reply apps/condo/domains/billing/access/SearchBillingReceiptsWithoutConsumerService.js]: As discussed with @pahaz we are going to responds with account data only

**Pull request closed without merging**